app/src/main/java/com/asfoundation/wallet/wallet_blocked/WalletBlockedInteract.kt

```diff
@@ -0,0 +1,18 @@
+package com.asfoundation.wallet.wallet_blocked
+
+import com.asfoundation.wallet.interact.FindDefaultWalletInteract
+import io.reactivex.Single
+import java.util.concurrent.TimeUnit
+
+class WalletBlockedInteract(
+    private val findDefaultWalletInteract: FindDefaultWalletInteract,
+    private val walletStatusRepository: WalletStatusRepository
+) {
+
+  fun isWalletBlocked(): Single<Boolean> {
+    return findDefaultWalletInteract.find()
+        .flatMap { walletStatusRepository.isWalletBlocked(it.address) }
+        .delay(1, TimeUnit.SECONDS)
```

Comment:
Why the delay?

Reply:
So the animation runs for a while instead of showing and disappearing